### PR TITLE
Connection pane improvements

### PR DESCRIPTION
**Pull request by skodman**
Created **brixter.env** environment to hold the data and reduce the number of API calls (if possible). I wanted to separate clusters by their purpose so I've added **skodman_test** element to the connections tree  list with two leafs under neath: Personal and Job (left Pools from the list as we are not using those so i can't test, should probably be be added). Have a look at the **brixter.env** environment,  **get_clusters_2** and **filter_clusters** functions below. Was wondering if this is something you might want to consider and if so i can cleanup the code.
I've also made some changes to fix the name of the workspace in the Connections tab.

### CODE REVIEW

File: R/connection-pane.R [hunk 1]
```diff
@@ -2,6 +2,11 @@
 # - can only return so many objects at once, how to paginate via UI?
 #   - impacts experiments, model registry, etc.
 
+# adding new environment to hold the data.
+brixter.env <- new.env()
```
Comment (skodman) [inline R/connection-pane.R:6]: Added **brixter.env** to hold the data which can then be filtered without API calls

File: R/connection-pane.R [hunk 2]
```diff
@@ -404,15 +416,46 @@ get_feature_table_columns <- function(table, host, token) {
...
+
     list(
       name = as.character(glue::glue("[{x$state}] {x$cluster_name} ({x$cluster_id})")),
       type = "cluster"
     )
   })
 }
 
+
+get_clusters_2 <- function(host, token) {
```
Comment (skodman) [inline R/connection-pane.R:431]: This function loads clusters into the brixter.env and just shows two leaves underneath. Those leaves are expanded on consecutive click when clusters are filtered from the brixter.env$cluster

File: R/connection-pane.R [hunk 3]
```diff
@@ -404,15 +416,46 @@ get_feature_table_columns <- function(table, host, token) {
...
+              name = as.character(glue::glue("[{x$state}] {x$cluster_name} ({x$cluster_id})")),
+              type = as.character(glue::glue("{x$type}"))
+            )
+
+
+          })
+
+}
+
+filter_clusters <-function(lFilter){
```
Comment (skodman) [inline R/connection-pane.R:449]: Filter selected branch of clusters

File: R/connection-pane.R [hunk 4]
```diff
@@ -483,11 +526,40 @@ list_objects <- function(host, token,
                          featuretable = NULL,
                          columns = NULL,
                          experiments = NULL,
+                         ### Skodman
+                         skodman_test=NULL,
                          ...) {
 
+  if (!is.null(skodman_test)) {
+    #browser()
+    if(is.null(brixter.env$clusters)){
```
Comment (skodman) [inline R/connection-pane.R:535]: Checks if clusters are already loaded into the brixter.env

File: R/connection-pane.R [hunk 5]
```diff
@@ -841,12 +916,14 @@ preview_object <- function(host, token, rowLimit,
 #' open_workspace(host = db_host(), token = db_token, name = "MyWorkspace")
 #' }
 #' @importFrom glue glue
-open_workspace <- function(host = db_host(), token = db_token(), name = NULL) {
+open_workspace <- function(host = db_host(), token = db_token(), name = db_wsid()) {
```
Comment (skodman) [inline R/connection-pane.R:919]: Fixes the name  for Connections tab

File: R/connection-pane.R [hunk 6]
```diff
@@ -925,9 +1005,13 @@ close_workspace <- function(host = db_host()) {
   if (!is.null(observer)) {
     observer$connectionClosed(type = "workspace", host = host)
   }
+
+  # clean up .rs.WorkingDataEnv
+  rm(list=ls(envir =brixter.env),envir=brixter.env)
```
Comment (skodman) [inline R/connection-pane.R:1010]: cleans up brixter.env variables

File: R/connection-pane.R [hunk 7]
```diff
@@ -995,7 +1079,32 @@ list_objects_types <- function() {
...
+        ))
+      )
+
+
     ))
   )
 }
+
+
+# clusters = list(contains = list(
```
Comment (skodman) [inline R/connection-pane.R:1105]: I'll remove this

File: R/connection-pane.R (hunk 5 above)
Comment (zacdav-db) [reply R/connection-pane.R:919]: I'd be open to making db_host() the default for `name` - that seems very sensible!

Comment (zacdav-db) [reply R/connection-pane.R:919]: Id's tend to be tricky for people to remember which is which

Comment (skodman) [reply R/connection-pane.R:919]: I agree, although the db_host(0 is equally confusing for me.  Maybe introducing workspace alias (which would be up to user to define in .databrickscfg) is the way to go. There user could define **test-ws**, **prod-ws** etc, something that is readable.

**Comment by zacdav-db** [comment]
Will close this for now and re-visit connection pane ideas once I've implemented some changes to it's core behaviour that are required.

**Pull request closed without merging**